lxd/containers_get.go

```diff
@@ -284,7 +352,13 @@ func doContainersGet(d *Daemon, r *http.Request) (interface{}, error) {
 		sort.Slice(resultList, func(i, j int) bool {
 			return resultList[i].Name < resultList[j].Name
 		})
-
+		if filterStr != "" {
+			intList := make([]interface{}, len(resultList))
+			for i := range resultList {
+			    intList[i] = resultList[i]
+			}
+			return doFilter(filterStr, intList), nil
+		}
```

Comment:
Instead of this, how about having functions in `lxd/instance` which take a list of []api.Instance or []api.InstanceFull and return a filtered set of those based on the filter (along with an error if the filter is invalid)?